src/components/UserProfile/UserProfile.js

```diff
@@ -50,31 +82,79 @@ const UserProfile = ({user}) => {
     (!followersVisible) ? setFollowersVisible(true) : setFollowersVisible(false)
   }
 
+  const renderTest = () => {
+    return (
+      <div>{!!GetVisitedUserInfo.data && !!GetFollowingInfo.data && renderProfile()}</div>
+    )
+  }
+
+  const checkFollowing = () => {
+    console.log('Users following the visited profile', GetFollowingInfo.data.usersFollowers)
+    // console.log('Current user id', GetFollowingInfo.data.usersFollowers)
+    let found = GetFollowingInfo.data.usersFollowers.find(user => user.id === value.toString())
+    console.log('Results', found)
+    if (found) {
+      return true
+    }
+    return false
+  }
+
+  const renderFollowBtn = () => {
+    return <button disabled={clicked} onClick={follow}>Follow</button>
+  }
+
+  const renderUnfollowBtn = () => {
+    return <button disabled={clicked} onClick={unfollow}>Unfollow</button>
+  }
+
+  const follow = () => {
+    setClicked(true)
+    // console.log('variables', {
+    //   'userId': value,
+    //   'friendId': GetVisitedUserInfo.data.user.id
+    // })
+    followUser({
+      variables: {
+        'userId': GetVisitedUserInfo.data.user.id,
+        'friendId': value
+      },
+    })
+  }
+
+  const unfollow = () => {
+    setClicked(true)
+    unFollowUser({
+      variables: {
+        'userId': GetVisitedUserInfo.data.user.id,
+        'followerId': value
+      },
+    })
+  }
```

Comment:
Love what you've built out here! Would we want to extract this into a few child components?